src/tools/gcc.lua

```diff
@@ -236,8 +236,20 @@
 
 	gcc.libraryDirectories = {
 		architecture = {
-			x86 = "-L/usr/lib32",
-			x86_64 = "-L/usr/lib64",
+			x86 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib32")
+				end
+				return r
+			end,
+			x86_64 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib64")
+				end
+				return r
+			end,
```

Comment:
Hello,
Using tables here was a bit of anticipation of what will finally be necessary some days. On my linux machine I have more than one arch-dependant path (`/usr/lib32`, `/usr/libx32`, `/usr/x86_64-linux-gnu/`, `/usr/arm-linux-gnueabi/` ...) and it may be different for other distros....

Moreover, there is currently no consistency in `config.mapFlags()` use. Some calls use strings, some use tables.

That was my reason to use tables here. However, if you still prefer strings, I will change it. Tell me